root/tree/index/runindex64.C

```diff
@@ -25,20 +27,22 @@ int runindex64(){
   tree->Branch("run", &run, "run/l");
   tree->Branch("event", &event, "event/l");
 
-  ULong64_t events[] = { 1,2,3, bigval, biguval, 5 };
-  run = 5;
+  ULong64_t   runs[] = { 8,5,5,5,      0,       0,  5,      4, 6, biguval,          1, maxMajor, maxMajor+1};
+  ULong64_t events[] = { 0,1,3,2, bigval, biguval,  5, bigval, 3,  bigval, maxMinor+1, maxMinor, 0};
   for(int i=0; i<sizeof(events)/sizeof(*events); i++){
+    run = runs[i];
     event = events[i];
     tree->Fill();
```

Comment:
With `events` being an `std::array` this simply becomes
```cpp
std::for_each(
    std::begin(events),
    std::end(events),
    [i=0] (ULong64_t val) mutable { run = runs[i]; event = val; tree->Fill(); i++;}
)
```

Reply:
On the other hand for this trivial example, it is not clear that the modern C++ is more 'readable' that the existing code.  More over, this change is not directly related to the goal of the PR.